packages/functional-tests/tests/subscription-tests/coupon.spec.ts

```diff
@@ -0,0 +1,54 @@
+import { test, expect } from '../../lib/fixtures/standard';
+
+test.describe('severity-1', () => {
+  test('subscribe with an expired coupon', async ({
+    pages: { relier, login, subscribe },
+  }, { project }) => {
+    test.skip(project.name === 'production', 'prod needs a valid credit card');
+    test.skip(project.name === 'local', 'No need to be run on local');
```

Comment:
Interesting. I wonder if instead of duplicating these `.skip()` conditions for each test in the suite individually, if we could use/abuse something like [`test.beforeEach()`](https://playwright.dev/docs/api/class-test#test-before-each) (or even [`test.beforeAll()`](https://playwright.dev/docs/api/class-test#test-before-all); both of which seem to have access to the [`TestInfo`](https://playwright.dev/docs/api/class-testinfo) object) and get the project info there.

In fact, [`test.skip()`](https://playwright.dev/docs/api/class-test#test-skip-3) does have a similar use case that shows that usage. 🤷

Reply:
https://playwright.dev/docs/test-annotations#conditionally-skip-a-group-of-tests also has a section for this which might reduce the boilerplate a bit.

Reply:
Actually, so far this was the only combination that worked for me for conditionally skipping an entire suite based on `project.name`/env:
```js
  test.beforeAll(({}, {project}) => {
    test.skip(project.name !== "stage", "Only run these tests in 'stage' env");
  });
```

```sh
> playwright test index.spec.js --headed

Running 3 tests using 3 workers

  ✓  1 [stage] › index.spec.js:11:3 › coupon suite › 12 month plan (9s)
  -  2 [production] › index.spec.js:11:3 › coupon suite › 12 month plan
  -  3 [local] › index.spec.js:11:3 › coupon suite › 12 month plan

  2 skipped
  1 passed (10s)
```